controller/state.go

```diff
@@ -448,6 +449,32 @@ func (m *appStateManager) CompareAppState(app *v1alpha1.Application, project *ap
 					LastTransitionTime: &now,
 				})
 			}
+
+			// Live namespaces which are managed namespaces (i.e. application namespaces which are managed with
+			// CreateNamespace=true and has non-nil managedNamespaceMetadata) will (usually) not have a corresponding
+			// entry in source control. In order for the namespace not to risk being pruned, we'll need to generate a
+			// namespace which we can compare the live namespace with. For that, we'll do the same as is done in
+			// gitops-engine, the difference here being that we create a managed namespace which is only used for comparison.
+			if liveObj.GetKind() == kubeutil.NamespaceKind && liveObj.GetName() == app.Spec.Destination.Namespace && app.Spec.SyncPolicy != nil && app.Spec.SyncPolicy.ManagedNamespaceMetadata != nil {
```

Comment:
I think that this part will also need to verify that a `targetObj` matching the namespace doesn't already exist